src/Network/Pusher/Data.hs

```diff
@@ -201,3 +226,136 @@ type Event = T.Text
 type EventData = T.Text
 
 type SocketID = T.Text
+
+-- | Up to 164 characters where each character is ASCII upper or lower case, a
+-- number or one of _=@,.;
+-- Note: hyphen - is NOT valid as it is reserved for the possibility of marking
+-- interest names with prefixes such as private- or presence-
+newtype Interest =
+  Interest T.Text
+
+mkInterest :: T.Text -> Maybe Interest
+mkInterest txt
+  | T.length txt <= 164 &&
+      T.all (\c -> isAlphaNum c || HS.member c permitted) txt =
+    Just . Interest $ txt
+  | otherwise = Nothing
+  where
+    permitted = HS.fromList "_=@,.;"
+
+instance A.FromJSON Interest where
+  parseJSON v =
+    case v of
+      A.String s ->
+        case mkInterest s of
+          Nothing ->
+            fail $
+            "An Interest contains invalid characters or is too long: " ++ show s
+          Just istr -> pure istr
+      _ -> failExpectStr v
+
+instance A.ToJSON Interest where
+  toJSON (Interest txt) = A.String txt
+
+-- | URL to which pusher will send information about sent push notifications
+type WebhookURL = T.Text
+
+-- | Level of detail sent to WebhookURL. Defaults to Info
+data WebhookLevel
+  = Info -- ^ Errors only
+  | Debug -- ^ Everything
+
+instance A.FromJSON WebhookLevel where
+  parseJSON v =
+    case v of
+      A.String s
+        | s == "INFO" -> pure Info
+        | s == "DEBUG" -> pure Debug
```

Comment:
As I mentioned last night, I think we don't really agree on the terms :P As for your final point, yes I agree!

Anyway I don't feel too strongly, so feel free to keep as-is.